### ISSUE TITLE
Scaling of non-integer sizes 

### ISSUE THREAD

**Issue by Sushisource** [ggez/ggez#608]
Hello! Noticed after upgrading to 0.5-rc1 that scaling behavior seems to have changed a bit. I draw little boxes composed of rects on each side, and those rects are all of uniform thickness. This width/height is `0.2` (meters, in-world-units) when the draw call is made, but also is scaled by a previous call to `graphics::set_transform` and `apply_transformation`

In 0.44 this always resulted in uniform thickness, but now they end up looking different depending on the dimensions of the boxes / positioning:

![example](https://i.imgur.com/Ll03F0a.png)

You can see the bottom edge is a bit chunky. 

Any idea why this changed? I can totally believe I'm doing something dumb with my scaling (like, should I not be trying to scale directly from world units in the draw call?), but the behavior is definitely different.

**Comment by icefoxen** [comment]
No idea off the top of my head, it shouldn't have changed.  Using `set_transform()` and such to make a convenient world-units coordinate system should be perfectly valid.  Does it alias around as you move the box up and down or side to side?

**Comment by Sushisource** [comment]
Indeed, it does a bit, here's a little recording:

![weirdaliasing](https://user-images.githubusercontent.com/517683/56111356-88bf5e80-5f0c-11e9-9003-dbf4fdb20188.gif)

**Comment by icefoxen** [comment]
Very nice test!  In that case I would recommend:

 * Turn off filtering of textures and multisampling if it's enabled
 * Make sure the coordinates you draw on always map to integer pixels with no fraction.

Not sure that ggez itself can do much else...  I need to come up with a nice example of how to implement something like this though.  It's something I've fought with myself and is only going to be a more common problem as time goes.

**Comment by icefoxen** [comment]
Any update on this?

**Comment by Sushisource** [comment]
@icefoxen Managed to get some time to try this out again today. Updated to rc2 first, and *poof* problem is gone. So, you fixed it! :tada: :tada:! Was there an intentional fix? I'd be curious to know what it was.

I did ponder a bit how one could ensure that, post-scaling, everything is drawn in integer sizes. Making sure that everything is clamped to integer size world coordinates wouldn't work since that'd be too low-rez (at least, not without going from meters to centimeters and that has a whole host of other issues). Perhaps a "snap to integer coords post-scaling" option on `DrawParam`, or maybe more generally, being able to provide some closure for manipulating coords & other options post-scaling?
